File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        hash = {} 
        for i, num in enumerate(nums):
            hash[num] = i
        
        for i, num in enumerate(nums):
            desired = target - num
            if desired in hash and hash[desired] != i:
                return i, hash[desired]
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> comps;
        for (int i = 0; i< nums.size(); i++){
            if(comps.find(target - nums[i]) != comps.end()){
                return {comps[target - nums[i]], i};
            }
            comps[nums[i]] = i;
        }
        return {};
    }
};